#Do-Now-5

## Wednesday, June 10, 2015

### Reviewing: dictionaries, lists, CSVs

 #### 1) What's the output of the following lines?

In [1]:
name = "Max"
print "hello", "name"
print "hello", name

hello name
hello Max


##### when the string in "", it is string
##### name is a vlue named "Max"

#### 2) What is the output of the following code? Which lines give you errors?

In [4]:
borough_name = 'Manhattan'
z = [ 'Manhattan', 'Queens' ]
x = { 'borough_name': 'Manhattan', 'population': 500 }
y = {'Manhattan': 500,'Queens': 200}

print x['borough_name']

Manhattan


In [ ]:
print x[borough_name]

##### borough_name is not defined. It should be like this print x['borough_name'].

In [6]:
print x[0]

KeyError: 0

#### when write the matching value, it should be the list, but x is dictionary.

In [7]:
print y['borough_name']

KeyError: 'borough_name'

##### when write the matching value, it should be the list, but x is dictionary.

In [8]:
print y[borough_name]

500


##### borough_name is defined as "Manhatann", then ,the value of Manhatann in y is 500.
##### so print out 500.

In [9]:
print y[0]

KeyError: 0

##### y is not a list, the dictionary is not in order.

In [10]:
print z['borough_name']

TypeError: list indices must be integers, not str

###### the list have no any intergers matching the string. so there is an error.

In [12]:
print z[borough_name]


TypeError: list indices must be integers, not str

##### It should be Z ['Manhattan']

In [13]:
print z[0]

Manhattan


##### z is a list, then the index of it is n-1, then it is 1, so the value matching is Manhattan

###### 3) We have a list called numbers, but we had an off-by-one error: add one to every element of the list, then sum the results. Do this in one line.

In [14]:
sum [(number+1) for number in numbers]

SyntaxError: invalid syntax (<ipython-input-14-c55af4a11085>, line 1)

##### 4a) Given the following, write code to calculate how many murders we have in total.

In [41]:
murders = { 'Albany': 23, 'Kings County': 10, 'Rochester': 7, 'Yonkers': 9}
total = 0
for murder in murders.values():
    total += murder
print total

49


#### 4b) Write code to calculate the percentage of the murders happened in Kings County.

In [2]:
murders = { 'Albany': 23, 'Kings County': 10, 'Rochester': 7, 'Yonkers': 9}
print float(murders['Kings County'])/float(sum(murders.values())) * 100,'%'

20.4081632653 %


##### In dictionary murders, the value'Kins County' is  10
##### murders.values() is 23, 10,7, 9 . the sum value is 49:
##### print 10/49 %,  it is 20.408%.